src/LEGO.AsyncAPI.Readers/JsonHelper.cs

```diff
@@ -12,7 +12,7 @@ internal static class JsonHelper
         public static string GetScalarValue(this JsonNode node)
         {
             var scalarNode = node is JsonValue value ? value : throw new AsyncApiException($"Expected scalar value");
-            return Convert.ToString(scalarNode.GetValue<object>(), CultureInfo.InvariantCulture);
+            return Convert.ToString(scalarNode.GetValue<object>(), Configuration.CultureInfo);
```

Comment:
instead of tying in, the cultureinfo via a static class.
Lets try to move it to 'readersettings' instead.
This is generally passed in the ParsingContext.
so it should be available for most of the node types.
And as these now hold the parsing logic (double.parse etc.) we could refactor all of these Parse calls to use the node.GetIntegerValue etc calls to utilize the reader settings for cultureinfo.

Reply:
Context is a public property of the base parsenode type.
So you should be able to get it out everywhere you have a node, actually.
So we might not even need to refactor anything.
Just use `n.Context.CultureInfo`

Reply:
Perfect, I can move it into there instead!

Reply:
Going down this path has made the review much bigger because there are a whole range of static help methods that I need to convert to either take the settings as a parameter or make instance.

Reply:
I think we can inline `jsonhelper` where ever its used, and get rid of it once and for all.
for JsonSchemaDeserializer, you already made changes to this.
Its simply a change from `Configuration.CultureInfo` to `n.Context.CultureInfo` instead.
Just be sure to pass the setting from ReaderSettings to the ParsingContext.

I like the idea of having a base AsyncApiSettings class, where CultureInfo would sit.

Reply:
This is going to require updating every single deserializer type as they are all currently static dictionary of actions.

Reply:
> This is going to require updating every single deserializer type as they are all currently static dictionary of actions.

It really shouldnt require this.
simply change `Configuration.CultureInfo` for `n.Context.CultureInfo` where applicable.
You already did most of the work - you just need to use the parsingContext instead.
I dont think we use cultureinfo in any of the deserializers outside of the ones mentioned.

The node already has the information, so you dont need to do any refactoring outside of what you've done already (with the replacement). And then pass the cultureinfo from basesettings to the parsingcontext (for reading).

Which writers actually use cultureinfo?

Reply:
Ive made a PR that shows what i mean here https://github.com/LEGO/AsyncAPI.NET/pull/159

Reply:
Yeah I missed that in the static helpers that there is the `n.context.Setting.CultureInfo` that could be used. That was the part that was tripping me up.

Reply:
Its not great. I dont really like the idea of having to dot all the way through the node, to do this kind of parsing.
Perhaps at a later stage we can add conversion deeper down, instead of relying on it, at the level we do.
But for now, itll do.